src/element/linearElementEditor.ts

```diff
@@ -359,15 +363,15 @@ export class LinearElementEditor {
     };
   }
 
-  static isHittingMidPoint = (
+  static getSegmentMidpointHitCoords = (
```

Comment:
this function must be optimized as it can take 150+ ms per frame for 50-point line (common in libraries) when called in the pointermove handler to determine which (mid)point you're hovering over — which causes noticeable lag, even race conditions where points tend to stay in hover state even after you move the cursor away.

If we can't make the algo better, e.g. narrow down to the segment the cursor is in instead of computing cursor distance to every line segment center, then we'll need to precalculate all segment midpoints and cache on line editor construction (it would also need to update on point move/delete/add/etc.).